include/alpaka/kernel/Traits.hpp

```diff
@@ -199,14 +199,23 @@ namespace alpaka
                 static_assert(std::is_same_v<Result, void>, "The TKernelFnObj is required to return void!");
             }
         };
+
+        // asserts that T is trivially copyable. We put this in a separate function so we can see which T would fail
+        // the test, when called from a fold expression.
+        template<typename T>
+        inline void assertKernelArgIsTriviallyCopyable()
+        {
+            static_assert(std::is_trivially_copyable_v<T>, "The kernel argument T must be trivially copyable!");
```

Comment:
```suggestion
            static_assert(std::is_empty<T>|| std::is_trivially_copyable_v<T>, "The kernel argument T must be trivially copyable!");
```

Reply:
I pushed a test with you suggested workaround

Reply:
Great, thx!